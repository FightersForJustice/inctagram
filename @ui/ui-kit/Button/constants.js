export const BUTTON_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  BASIC: 'Basic',
  OUTLINED: 'Outlined',
  GHOST: 'Ghost',
})

export const BUTTON_VARIATIONS = Object.freeze({
  FULL_WIDTH: 'FullWidth',
  AUTO_HEIGHT: 'AutoHeight',
  DEFAULT: 'Default',
})
