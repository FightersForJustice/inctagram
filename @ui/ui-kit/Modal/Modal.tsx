import classNames from "classnames"
import s from "./modal.module.scss"

type ModalType = {
    active: boolean
    setActive: any
    title: string
    children: any
  }

const Modal: React.FC<ModalType> = ({active, setActive, title, children}) => {
    const open = active ? classNames(s.modal, s.effect, s.show) : classNames(s.modal, s.effect)  
    return (
    <>
        <div className={open} >
            <div className={s.content}>
                <div className={s.item}>
                    <div className={s.title}>{title}t</div>
                    <button className={s.buttonClose} onClick={() => {setActive(false)}}>
                        <img src="/icons/close.svg" alt="" />
                    </button>
                </div>
                <div className={s.contents}>{children}</div>
            </div>
        </div>
        <div className={s.overlay}></div>
    </>
    )
}
export default Modal;