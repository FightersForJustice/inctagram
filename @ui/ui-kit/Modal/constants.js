export const CLOSE_MODAL_TEST_ID = 'closeModal'
