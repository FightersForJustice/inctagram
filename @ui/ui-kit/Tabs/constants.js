export const TAB_COLORS = Object.freeze({
  PRIMARY: 'Primary',
  GHOST: 'Ghost',
})
export const profileSettingsData = [
  { value: 'general', label: 'General information' },
  { value: 'Devices', label: 'Devices' },
  { value: 'AccountManagment', label: 'Account Managment' },
  { value: 'Payments', label: 'MyPayments' },
]
