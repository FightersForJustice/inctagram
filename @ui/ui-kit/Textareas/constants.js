export const TEXTAEREA_COLORS = Object.freeze({
  DEFAULT: 'Default',
  ACTIVE: 'Active',
  ERROR: 'Error',
})
