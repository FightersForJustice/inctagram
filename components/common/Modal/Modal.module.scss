@import url('https://fonts.googleapis.com/css2?family=Inter:wght@100;200;300;400;500;600;700;800;900&display=swap'); /*  ?  */
.modal {
  position: fixed;
  top: 0;
  right: 0;
  bottom: 0;
  left: 0;
  background-color: #00000040;
  z-index: 1;
  .modalBody {
    font-family: Inter;
    color: #fff;
    position: fixed;
    background-color: #333;
    top: 0;
    right: 0;
    bottom: 0;
    left: 0;
    display: block;
    margin: auto;
    height: fit-content;
    max-width: 400px;
    border-radius: 2px;
    border: 1px solid #4c4c4c;
    .item {
      border: 1px solid #4c4c4c;
      display: flex;
      justify-content: space-between;
      padding: 15px 24px;
      .title {
        font-size: 20px;
        font-weight: 400;
      }
      .buttonClose {
        border: 0;
        background-color: #333;
        cursor: pointer;
      }
    }
    .content {
      padding: 24px;
      .text {
        font-weight: 300;
        line-height: 24px;
      }
      .buttonBloc {
        text-align: right;
        padding-top: 24px;
        .button {
          border-radius: 2px;
          background: #397df6;
          border: 0;
          color: #fff;
          font-family: Inter;
          font-weight: 600;
          padding: 10px 24px;
          cursor: pointer;
        }
      }
    }
  }
}
