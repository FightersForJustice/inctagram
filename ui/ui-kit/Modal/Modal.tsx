import s from "./modal.module.scss"

type ModalType = {
    active: boolean
    setActive: any
    title: string
    children: any
  }

const Modal: React.FC<ModalType> = ({active, setActive, title, children}) => {
    const classs = `${s.modal} ${s.effect}`
    const open = active ? `${s.modal} ${s.effect} ${s.show}` : classs  
    return (
    <>
        <div className={open} >
            <div className={s.content}>
                <div className={s.item}>
                    <div className={s.title}>{title}t</div>
                    <button className={s.buttonClose} onClick={() => {setActive(false)}}>
                        <img src="/icons/close.svg" alt="" />
                    </button>
                </div>
                <div className={s.contents}>{children}</div>
            </div>
        </div>
        <div className={s.overlay}></div>
    </>
    )
}
export default Modal;