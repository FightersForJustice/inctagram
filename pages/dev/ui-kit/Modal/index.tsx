import { PageWrapper } from 'components/PageWrapper/PageWrapper'
import { getLayout } from '@/components/Layout/Layout'
import Modal from '@/ui/ui-kit/Modal/Modal'
import { useState } from 'react'
import style from './index.module.scss'

export const getStaticProps = async () => {
  return {
    props: {},
  }
}

const Modals = () => {
    const [ModalActive, setModalActive] = useState(false);
  return (
    <PageWrapper>
        <button className={style.openModal} onClick={() => {setModalActive(true)}}>Modal open</button>
        <Modal active={ModalActive} setActive={setModalActive} title="Email sent">
          <div className={style.text}>We have sent a link to confirm your email to epam@epam.com</div>
          <div className={style.buttonBloc}>
            <button className={style.button}>
              OK
            </button>
          </div>
        </Modal>
    </PageWrapper>
  )
}

Modals.getLayout = getLayout
export default Modals