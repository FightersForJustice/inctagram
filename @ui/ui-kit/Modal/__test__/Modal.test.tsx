import React from 'react'
import { render, screen, fireEvent, cleanup } from '@testing-library/react'
import Modal from '../Modal'
import { CLOSE_MODAL_TEST_ID } from '../constants'

const title = 'Test Modal'
const children = <div>Some Text</div>
const setActive = jest.fn()

const activeProps = [true, false]
const closeProps = [true, false]

describe('render snapshot correctly \t', () => {
  afterEach(cleanup)

  activeProps.forEach((active) => {
    closeProps.forEach((close) => {
      it(`active:${active} \t
      setActive: ${setActive} \t
      title: ${title} \t
      close: ${close} \t
            `, () => {
        const props = {
          active: active,
          setActive: setActive,
          title: title,
          close: close,
        }
        const { asFragment } = render(<Modal {...props}>{children}</Modal>)

        expect(asFragment()).toMatchSnapshot()
      })
    })
  })
})

describe('Modal Component', () => {
  afterEach(cleanup)

  it('renders correctly with title and children', () => {
    render(
      <Modal active={true} setActive={() => {}} title={title}>
        {children}
      </Modal>
    )

    expect(screen.getByText(title)).toBeInTheDocument()
    expect(screen.getByText('Some Text')).toBeInTheDocument()
  })

  it('closes when close button is clicked', () => {
    render(
      <Modal active={true} setActive={setActive} title={title} close={true}>
        {children}
      </Modal>
    )

    const closeButton = screen.getByTestId(CLOSE_MODAL_TEST_ID)
    fireEvent.click(closeButton)
    expect(setActive).toHaveBeenCalledWith(false)
  })

  it('does not show close button when "close" prop is not provided', () => {
    render(
      <Modal active={true} setActive={() => {}} title={title}>
        {children}
      </Modal>
    )

    expect(screen.queryByTestId('close-button')).not.toBeInTheDocument()
  })

  it('renders correctly when children is null', () => {
    render(
      <Modal active={true} setActive={() => {}} title={title}>
        {null}
      </Modal>
    )

    expect(screen.getByText(title)).toBeInTheDocument()
  })
})
