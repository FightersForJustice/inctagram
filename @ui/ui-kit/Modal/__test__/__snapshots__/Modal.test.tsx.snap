// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render snapshot correctly 	 active:false 	
      setActive: function () {
        return fn.apply(this, arguments);
      } 	
      title: Test Modal 	
      close: false 	
             1`] = `
<DocumentFragment>
  <div
    class="modal effect"
  >
    <div
      class="content"
    >
      <div
        class="item"
      >
        <div
          class="title"
        >
          Test Modal
        </div>
      </div>
      <div
        class="contents"
      >
        <div>
          Some Text
        </div>
      </div>
    </div>
  </div>
  <div
    class="overlay"
  />
</DocumentFragment>
`;

exports[`render snapshot correctly 	 active:false 	
      setActive: function () {
        return fn.apply(this, arguments);
      } 	
      title: Test Modal 	
      close: true 	
             1`] = `
<DocumentFragment>
  <div
    class="modal effect"
  >
    <div
      class="content"
    >
      <div
        class="item"
      >
        <div
          class="title"
        >
          Test Modal
        </div>
        <button
          class="buttonClose"
          data-testid="closeModal"
        >
          <img
            alt=""
            src="/icons/close.svg"
          />
        </button>
      </div>
      <div
        class="contents"
      >
        <div>
          Some Text
        </div>
      </div>
    </div>
  </div>
  <div
    class="overlay"
  />
</DocumentFragment>
`;

exports[`render snapshot correctly 	 active:true 	
      setActive: function () {
        return fn.apply(this, arguments);
      } 	
      title: Test Modal 	
      close: false 	
             1`] = `
<DocumentFragment>
  <div
    class="modal effect show"
  >
    <div
      class="content"
    >
      <div
        class="item"
      >
        <div
          class="title"
        >
          Test Modal
        </div>
      </div>
      <div
        class="contents"
      >
        <div>
          Some Text
        </div>
      </div>
    </div>
  </div>
  <div
    class="overlay"
  />
</DocumentFragment>
`;

exports[`render snapshot correctly 	 active:true 	
      setActive: function () {
        return fn.apply(this, arguments);
      } 	
      title: Test Modal 	
      close: true 	
             1`] = `
<DocumentFragment>
  <div
    class="modal effect show"
  >
    <div
      class="content"
    >
      <div
        class="item"
      >
        <div
          class="title"
        >
          Test Modal
        </div>
        <button
          class="buttonClose"
          data-testid="closeModal"
        >
          <img
            alt=""
            src="/icons/close.svg"
          />
        </button>
      </div>
      <div
        class="contents"
      >
        <div>
          Some Text
        </div>
      </div>
    </div>
  </div>
  <div
    class="overlay"
  />
</DocumentFragment>
`;
